hack/generators/kic/kustomize.go

```diff
@@ -0,0 +1,62 @@
+package kic
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// BuildKustomizeForURLAndRef runs kustomize build for the provided URL and ref.
+// It returns the output of the kustomize build command.
+func BuildKustomizeForURLAndRef(ctx context.Context, url, ref string) ([]byte, error) {
+	const kustomizationFileName = "kustomization.yaml"
+	tmpDir, err := os.MkdirTemp("", fmt.Sprintf("kustomize-kic-%s", ref))
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	kustomizeResourceURL := fmt.Sprintf("%s?ref=%s", url, ref)
+	kustomization := `` +
+		`apiVersion: kustomize.config.k8s.io/v1beta1` + "\n" +
+		`kind: Kustomization` + "\n" +
+		`resources:` + "\n" +
+		`- ` + kustomizeResourceURL + "\n"
+
+	path := filepath.Join(tmpDir, kustomizationFileName)
+	if err := os.WriteFile(path, []byte(kustomization), 0o644); err != nil {
+		return nil, err
+	}
```

Comment:
Couldn't this be simplified to running `kustomize build https://github.com/kong/kubernetes-ingress-controller/config/webhook\?ref=v3.2.0` command instead?

Reply:
Looks like it can be done this way 👍 f610301a5406f18fec6f401b1a1a830f095ae482

PTAL